rocks-lib/src/build/mod.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{
+    config::Config,
+    rockspec::{Build, BuildBackendSpec, RockSourceSpec, Rockspec},
+    tree::Tree,
+};
+use eyre::Result;
+use git2::Repository;
+
+pub fn build(rockspec: Rockspec, config: &Config, no_install: bool) -> Result<()> {
+    // TODO(vhyrro): Use a more serious isolation strategy here.
+    let temp_dir = tempdir::TempDir::new(&rockspec.package)?;
+
+    let previous_dir = std::env::current_dir()?;
+
+    // Install the source in order to build.
+    match &rockspec.source.current_platform().source_spec {
+        RockSourceSpec::Git(git) => {
+            let repo = Repository::clone(&git.url.to_string(), &temp_dir)?;
+
+            if let Some(commit_hash) = &git.checkout_ref {
+                let (object, _) = repo.revparse_ext(commit_hash)?;
+                repo.checkout_tree(&object, None)?;
+            }
+        }
+        RockSourceSpec::Url(_) => todo!(),
+        RockSourceSpec::File(_) => todo!(),
+        RockSourceSpec::Cvs(_) => unimplemented!(),
+        RockSourceSpec::Mercurial(_) => unimplemented!(),
+        RockSourceSpec::Sscm(_) => unimplemented!(),
+        RockSourceSpec::Svn(_) => unimplemented!(),
+    };
+
+    std::env::set_current_dir(&temp_dir)?;
+
+    // TODO(vhyrro): Instead of copying bit-by-bit we should instead perform all of these
+    // operations in the temporary directory itself and then copy all results over once they've
+    // succeeded.
+
+    let tree = Tree::new(
+        &config.tree,
+        config
+            .lua_version
+            .as_ref()
+            .unwrap_or(&crate::config::LuaVersion::Lua51),
```

Comment:
`lua_version` should be read from a configuration file instead of defaulting to Lua 5.1 here :)

Reply:
Might also be nice to be able to set it from an environmen variable